Feat: Obfuscated Link Key - 3001

hamed-valiollahi opened the pull request:
This PR implements an obfuscated link key for secure anonymous access to organization forms, including key management, validation, and auditing.  

Closes #3001

AlexZorkin commented on backend/lcfs/web/api/forms/views.py:
```diff
@@ -0,0 +1,114 @@
+    """
+    Get form by slug for authenticated users (BCeID users)
+    """
+    logger.info(
```
don't think we need this logger here. Error logging is already handled, this would be extraneous.

AlexZorkin commented on backend/lcfs/web/api/forms/views.py:
```diff
@@ -0,0 +1,114 @@
+
+    organization_name = None
+    # Check if we have a link key user from the state
+    link_key_user = getattr(request.state, "user", None)
```
this seems a bit weird, we should just use our request.user directly. Do you see a case for having this getattr? Does it error out from an anonymous api call?

AlexZorkin commented on backend/lcfs/web/application.py:
```diff
@@ -71,12 +71,21 @@ async def authenticate(self, request):
+        if path.startswith("/api/forms/"):
+            path_parts = path.split("/")
+            # If it has 5 parts: ['', 'api', 'forms', 'form_slug', 'link_key']
+            if len(path_parts) == 5 and path_parts[4]:  # Has link key
```
Hmm this this is a bit fragile because of the positional assumptions.  Maybe some form of regex? 
FORM_LINK_RE = re.compile(
    r"^/api/forms/(?P<form_slug>[^/]+)/(?P<link_key>[A-Za-z0-9_-]{16,})/?$"
)
If we continue to add public routes i think we should look at creating a public sub app instead.

hamed-valiollahi replied:
For sure! I agree with you.

AlexZorkin commented on frontend/src/utils/clipboard.js:
This is a very useful utils class. Where else could we use this in the app?

hamed-valiollahi replied:
Thanks, Alex. I checked the other places where we have the copy feature and replaced them with this util.

AlexZorkin commented on frontend/src/views/Forms/FormView.jsx:
```diff
@@ -0,0 +1,173 @@
+          axiosConfig.headers = {
+            Authorization: `Bearer ${keycloak.token}`
+          }
+          console.log(
```
should remove this logging

AlexZorkin commented on frontend/src/views/Forms/FormView.jsx:
```diff
@@ -0,0 +1,173 @@
+          )
+        }
+
+        console.log('API URL:', apiUrl)
```
same here

AlexZorkin commented on frontend/src/views/Forms/FormView.jsx:
```diff
@@ -0,0 +1,173 @@
+        setFormData(response.data)
+      } catch (err) {
+        console.error('Error fetching form data:', err)
+        console.error('Error response:', err.response)
```
cleanup logging

AlexZorkin commented on frontend/src/views/Forms/FormView.jsx:
```diff
@@ -0,0 +1,173 @@
+        console.error('Error data:', err.response?.data)
+
+        let errorMessage = 'Failed to load form'
+        if (err.response?.data?.detail) {
```
the if else logic gets a little complicated here. Please clean it up with a status to message mapping method.

AlexZorkin commented on frontend/src/views/Organizations/OrganizationView/components/LinkKeyManagement.jsx:
The number of methods in here could be moved to a service class.

AlexZorkin reviewed (changes requested):
Looking good, just some comments to address. Then some testing files and we're set. 👍

hamed-valiollahi commented:
> Looking good, just some comments to address. Then some testing files and we're set. 👍

Thanks so much, Alex. I’ve addressed all your comments and added tests.
